custom_enumerable_methods.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    my_self = []
+    i = 0
+    while i < my_self.size
+      yield(my_self[i])
+      i += 1
+    end
+    my_self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_self = []
+    (i...my_self.size).my_each do |i|
+      yield(my_self[i], i)
+    end
+    my_self
+  end
+
+  def my_select
+    result = []
+    my_self = []
+    my_self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
+
+  def my_all?
+    result = true
+    my_self = []
+    my_self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_self = []
+    my_self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_self = []
+    my_self.my_each do |i|
+      result = false if my_self[i] == yield(i)
+    end
+    result
+  end
+
+  def my_count
+    i = 0
+    my_self = []
+    my_self.my_each { i += 1 }
+    i
+  end
+
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    new_self = []
+    my_each do |element|
+      new_self << if is_a? Hash
+                    yield(element[0], element[1])
+                  else
+                    yield(element)
+                  end
+    end
+    new_self
+  end
+
+  def my_inject(initial = 0)
+    i = 0
+    my_self = self
+    accumulator = initial
+    while i < my_self.length
+      accumulator = yield(accumulator, my_self[i])
+      i += 1
+    end
+    accumulator
+  end
```

Comment:
```rb
2.6.5 :146 > range = Range.new(0,50)
2.6.5 :147 > actual = range.my_inject(4) { |prod, n| prod * n }
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        3: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        2: from (irb):147
        1: from custom_enumerable_methods.rb:79:in `my_inject'
NoMethodError (undefined method `length' for 0..50:Range)
2.6.5 :148 > expected = range.inject(4) { |prod, n| prod * n }
2.6.5 :149 > expected
 => 0 
2.6.5 :150 >
```

Reply:
```rb
2.6.5 :152 > array
 => [5, 4, 4, 0, 5, 0, 3, 4, 0, 0, 2, 6, 5, 6, 5, 6, 2, 1, 3, 0, 8, 0, 0, 1, 7, 1, 3, 5, 6, 1, 1, 1, 6, 5, 7, 7, 2, 0, 3, 1, 7, 0, 1, 4, 5, 7, 6, 5, 4, 3, 6, 5, 4, 0, 6, 2, 7, 2, 5, 6, 4, 7, 1, 3, 6, 3, 3, 2, 6, 1, 7, 7, 8, 6, 1, 3, 5, 0, 4, 1, 0, 5, 4, 2, 5, 2, 5, 6, 6, 0, 1, 8, 4, 5, 4, 0, 4, 0, 8, 6] 
2.6.5 :153 > array.my_inject(:+)
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        3: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        2: from (irb):153
        1: from custom_enumerable_methods.rb:80:in `my_inject'
LocalJumpError (no block given (yield))
2.6.5 :154 > array.inject(:+)
 => 364 
2.6.5 :155 >
```

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more and fix the things.